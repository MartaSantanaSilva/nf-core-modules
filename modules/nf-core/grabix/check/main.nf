process GRABIX_CHECK {
    label 'process_low'

    conda "${moduleDir}/environment.yml"
    container "${ workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container ?
        'https://depot.galaxyproject.org/singularity/grabix:0.1.8--hdcf5f25_9':
        'biocontainers/grabix:0.1.8--hdcf5f25_9' }"

    input:
    path(input)

    output:
    env COMPRESS_BGZIP  , emit: compress_bgzip
    path "versions.yml" , emit: versions

    when:
    task.ext.when == null || task.ext.when

    script:
    def args = task.ext.args ?: ''

    """
    COMPRESS_BGZIP=\$(grabix check ${input})

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        grabix: \$(echo \$(grabix | sed -n -E 's/version: (.*)/\1/p'))
    END_VERSIONS
    """
}
