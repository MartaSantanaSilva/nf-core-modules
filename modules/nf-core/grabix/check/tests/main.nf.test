nextflow_process {

    name "Test Process GRABIX_CHECK"
    script "../main.nf"
    process "GRABIX_CHECK"
    tag "modules"
    tag "modules_nfcore"
    tag "grabix"
    tag "grabix/check"

    test("sarscov2") {
        when {
            process {
                """                
                input[0] = Channel.of([file(params.test_data_base + "/data/genomics/sarscov2/genome/genome.fasta.gz", checkIfExists: true)])
  
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out.compress_bgzip) },
                { assert snapshot(process.out.versions).match("versions") }
            )
        }

    }

}
