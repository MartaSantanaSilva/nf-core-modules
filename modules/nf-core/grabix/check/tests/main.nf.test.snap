{
    "versions": {
        "content": [
            [
                "versions.yml:md5,d68096f2e39f046ced7b8fb4672cae69"
            ]
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "23.10.1"
        },
        "timestamp": "2024-03-20T11:58:10.26615"
    }
}